### PR TITLE
server: Bump GHC version to 8.10.2

### PR DESCRIPTION
**Pull request by lexi-lambda**
This bumps GHC to version 8.10.2, pulling in a few patches that, among other things, should hopefully improve compiler performance. This is just internal housekeeping; we do not expect this to result in any observable difference in `graphql-engine` behavior or performance.

### CODE REVIEW

File: .circleci/config.yml
Comment (tirumaraiselvan) [inline .circleci/config.yml:6]: @lexi-lambda @jberryman BTW, both of you have push access to hasura repo in dockerhub :D

Screen Shot 2020-08-26 at 8 42 29 PM

Comment (lexi-lambda) [reply .circleci/config.yml:6]: Yes, I intend to push the image to the hasura org before merging.

**Review by tirumaraiselvan — APPROVED** [review]

**Pull request merged**
